pkg/sbproxy/reconcile/reconcile_visibilities.go

```diff
@@ -134,33 +157,63 @@ func (r *ReconciliationTask) getPlatformVisibilitiesFromCache() []*platform.Serv
 	return nil
 }
 
-func (r *ReconciliationTask) loadPlatformVisibilities(plans []*types.ServicePlan) ([]*platform.ServiceVisibilityEntity, error) {
+func (r *ReconciliationTask) loadPlatformVisibilitiesByBrokers(brokers []platform.ServiceBroker) ([]*platform.ServiceVisibilityEntity, error) {
 	logger := log.C(r.runContext)
 	logger.Debug("ReconciliationTask getting visibilities from platform")
 
-	platformVisibilities, err := r.platformClient.Visibility().GetVisibilitiesByPlans(r.runContext, plans)
+	visibilities, err := r.platformClient.Visibility().GetVisibilitiesByBrokers(r.runContext, brokers)
```

Comment:
you are passing SM brokers with SM broker IDs to a platform client. This requires special knowledge from the client on how to construct special names which only sbproxy-core needs. ProxyBrokerPrefix should not be used be used outside smproxy-core and IMO should not be exporeted

Reply:
You are right, this should be reworked